src/metabase/api/permissions.clj

```diff
@@ -62,14 +66,15 @@
 
 (defn- ordered-groups
   "Return a sequence of ordered `PermissionsGroups`, including the `MetaBot` group only if MetaBot is enabled."
-  [limit offset]
+  [group-filter limit offset]
   (db/select PermissionsGroup
              (cond-> {:where    (if (metabot/metabot-enabled)
                                   true
                                   [:not= :id (u/get-id (group/metabot))])
-                      :order-by [:%lower.name]}
-               (some? limit)  (hh/limit  limit)
-               (some? offset) (hh/offset offset))))
+                      :order-by [[:id :desc]]}
+               (some? group-filter) (hh/merge-where [:like :name (str "%" group-filter "%")])
+               (some? limit)        (hh/limit  limit)
```

Comment:
Do you need to wrap everything in `some?` here? The only place it makes a difference is when the value is false. e.g. if `x` is `false`, `x` itself is falsey while `(some? x)` is `true`. `some?` just checks whether something is not `nil`. So when people see `some?` they usually think you explicitly want to treat the value `false` as truthy -- there's no other reason to use it.

I don't see any situation where `limit` or `offset` would be `false`; even if they were, it seems like the wrong logic to say "`limit` is `false`, let's add it to the HoneySQL form anyway with `hh/limit`.